cpp/ycm/DLLDefines.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2015 ycmd contributors
+//
+// This file is part of YouCompleteMe.
+//
+// YouCompleteMe is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// YouCompleteMe is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with YouCompleteMe.  If not, see <http://www.gnu.org/licenses/>.
+
+#ifndef DLLDEFINES_H_0IYA3AQ3
+#define DLLDEFINES_H_0IYA3AQ3
+
+// We need to export symbols for gmock tests on Windows
+#if defined( _WIN32 ) && defined( TESTS_EXPORTS )
+  #define TESTS_EXPORT __declspec(dllexport)
```

Comment:
Maybe TESTS_EXPORTS is not the best name; it's saying what the symbols are used for instead of what the macro does.

Possibly use DLL_EXPORT instead?